### PR TITLE
Do not try to connect to models in error state.

### PR DESCRIPTION
**Pull request by frankban**
When initializing the app, filter models so that only the ones without errors are considered.
Also removed a duplicated check for no models with consequent switching to disconnected state.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6019/
Test PASSed.

**Comment by bac** [comment]
:+1:

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -743,15 +743,7 @@ YUI.add('juju-gui', function(Y) {
...
-          this.set('environmentList', modelList);
-          // If the modelList has no models in it then we have to drop the
-          // user into an uncommitted state.
-          if (modelList.length === 0) {
-            // Drop the user into the uncommitted state.
-            console.log('No models available, using unconnected mode.');
-            this.switchEnv();
-            return;
-          }
+          modelList = modelList.filter(model => !model.err);
```
Comment (urosj) [inline jujugui/static/gui/src/app/app.js]: But now you're not dropping the user in uncommited state anymore?

Comment (frankban) [inline jujugui/static/gui/src/app/app.js]: We are doing that later, this code here was a repetition.

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**